### PR TITLE
Feature/receive atmega payload

### PR DESCRIPTION
**Pull request by MCLEANS**
## Description
1.  Receive DHT, GPS and PMS values from ATMEGA328P.
2. Parse received data for debug and printing to web server.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: airrohr-firmware/airrohr-firmware.ino [hunk 1]
```diff
@@ -2679,6 +2688,62 @@ static void fetchSensorDHT(String& s) {
...
+			int last_character = dht_data.lastIndexOf(",");
+			s = dht_data.substring(0,(last_character+1));
+			parseDHTPayloadForDebug(s);
+			//Serial.println(s);
+		}
+	}
+
+	debug_outln_info(FPSTR(DBG_TXT_SEP));
+	debug_outln_verbose(FPSTR(DBG_TXT_END_READING), FPSTR(SENSORS_DHT22));
+	return s;
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino:2744]: @MCLEANS let's have this variable defined outside of the while loop to enable catching errors incase atmega is not available after the delay.

File: airrohr-firmware/airrohr-firmware.ino [hunk 2]
```diff
@@ -3066,6 +3131,74 @@ static void fetchSensorPMS(String& s) {
...
+				int last_character = pms_data.lastIndexOf(",");
+				s = pms_data.substring(0,(last_character+1));
+				parsePMSPayloadForDebug(s);
+				//Serial.println(s);
+			}
+		}
+	}
+
+	debug_outln_verbose(FPSTR(DBG_TXT_END_READING), FPSTR(SENSORS_PMSx003));
+	return s;
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino:3199]: @MCLEANS same here for the event that atmega is not available.

File: airrohr-firmware/airrohr-firmware.ino [hunk 3]
```diff
@@ -3066,6 +3131,74 @@ static void fetchSensorPMS(String& s) {
...
+ *****************************************************************/
+String fetchSensorPMSFromAtmega(){
+	debug_outln_verbose(FPSTR(DBG_TXT_START_READING), FPSTR(SENSORS_PMSx003));
+	RESERVE_STRING(s,SMALL_STR);
+
+	if(send_now){
+		atmega328p.println("fetchSensorPMS");
+		delay(WARMUPTIME_SDS_MS + READINGTIME_SDS_MS + 2000);
+		while(atmega328p.available() > 0){
+			String pms_data = atmega328p.readString();
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino:3188]: @MCLEANS do you have an example of what string is returned here?

File: airrohr-firmware/airrohr-firmware.ino [hunk 4]
```diff
@@ -3436,6 +3569,99 @@ static void fetchSensorGPS(String& s) {
...
+		if(gps_data.indexOf("GPS")){
+			int last_character = gps_data.lastIndexOf(",");
+			s = gps_data.substring(0,(last_character+1));
+			parseGPSPayloadForDebug(s);
+			//Serial.println(s);
+		}
+	}
+
+	debug_outln_info(FPSTR(DBG_TXT_SEP));
+	return s;
```
Comment (chegejames) [inline airrohr-firmware/airrohr-firmware.ino:3662]: @MCLEANS same here.

**Review by chegejames — CHANGES_REQUESTED** [review]

File: airrohr-firmware/airrohr-firmware.ino (hunk 1 above)
Comment (MCLEANS) [reply airrohr-firmware/airrohr-firmware.ino:2744]: @chegejames  The variable is defined outised the loop as "RESERVE_STRING(s,SMALL_STR)" , second line inside the fuction.

File: airrohr-firmware/airrohr-firmware.ino (hunk 3 above)
Comment (MCLEANS) [reply airrohr-firmware/airrohr-firmware.ino:3188]: @chegejames 
pms_payload

File: airrohr-firmware/airrohr-firmware.ino (hunk 1 above)
Comment (chegejames) [reply airrohr-firmware/airrohr-firmware.ino:2744]: @MCLEANS my bad I missed that.

**Review by chegejames — APPROVED** [review]

**Pull request merged**